admin_solutions/authentication.adoc

```diff
@@ -5,10 +5,308 @@ toc::[]
+and paste it to replace the existing stanza.
+. Make the following modifications to the `*identityProviders*` stanza:
+.. Change the provider `*name*` ("my_request_header_provider") to something
+unique and relevant to your deployment. This name is prefixed to the returned
+user ID to form an identity name.
+.. If required,
+xref:../install_config/configuring_authentication.adoc#mapping-identities-to-users[change `*mappingMethod*`] 
+to control how mappings are established between the provider's identities and
+user objects.
+.. Set the `*challenge*` parameter to *true* to redirect unauthenticated requests from clients expecting `WWW-Authenticate` challenges. 
+.. Set the `*provider.challengeURL*` parameter to the proxy URL to which to send clients expecting `WWW-Authenticate` challenges. 
+This parameter can include
+the 
+xref:../install_config/configuring_authentication.adoc#RequestHeaderIDP-urlquerytokens[`${url}` and `${query}` tokens]
+in the query portion of the URL. 
+.. Set the `*login*` parameter to *true* to redirect unauthenticated requests from clients expecting login flows. 
+.. Set the `*provider.loginURL*` parameter to the proxy URL to which to send clients expecting login flows. This parameter can include the 
+xref:../install_config/configuring_authentication.adoc#RequestHeaderIDP-urlquerytokens[`${url}` and `${query}` tokens]
+in the query portion of the URL. 
+.. Optionally, change the `*clientCA*` parameter to check incoming requests for a valid client certificate before the request's headers are checked for a user name. 
++
+[WARNING]
+====
+If you expect unauthenticated requests to reach the OAuth server, a `*clientCA*`
+parameter should be set for this identity provider. Otherwise, any direct
+request to the OAuth server can impersonate any identity from this provider,
+merely by setting a request header.
+====
+. Save your changes and close the file.
+. Start the {product-title} API server, specifying the configuration file you just modified:
++
+----
+$ openshift start master --config=<path/to/modified/config>/master-config.yaml
+----
+
+Once configured, any user logging in to the {product-title} web console will be
+prompted to log in using Request Header credentials.
+
+[[create-users-request-header-auth]]
+=== Creating Users with Request Header Authentication
+
+You do not create users in {product-title} when integrating with an external
+authentication provider, such as, in this case, Request Header. The Request
+Header authentication server is the system of record, meaning that users are
+defined there, and any user with a valid user name for the configured
+authentication server can log in.
+
+To add a user to {product-title}, the user must exist on the Request Header authentication server, and if required you must create a new Request Header authentication account for the user.
```

Comment:
This has to be more vague... they'll need to add the users to whatever system the proxy is using to authenticate them.